### DISCUSSION TITLE
Translations

### DISCUSSION THREAD

**Discussion by beyerenrico** [favware/graphql-pokemon#785] [General]
Hello!

I really like this repo, but I wondered if there is any way to translate the names?

**Comment by favna** [comment]
Right now this repo is not set up to allow for translations. Quite a few refactors would be needed to support them. Taking [items.ts](https://github.com/favware/graphql-pokemon/blob/a406fb7a02ed0b93373e073a8ef9d763a8e2e344/src/lib/assets/items.ts#L8-L13) for example to support translations the `name` and `description` for each item would have to be moved to to `src/lib/translations/{{language}}/items.json` and in `items.ts` the fields would probably have to have string constants.

And then the same for all other files in `src/lib/assets`

Once that is done the mappers in `src/lib/mappers` would have to be updated to get the translated strings, and `src/lib/resolvers` would need to be updated to accept a language string variable.

If someone is willing to make a pull request for this to at least make a start that would be very much appreciated as I lack time to in general, and even more so with Scarlet and Violet with right around the corner which means adding a whole bulk of data again.
